File: src/translator/zh2en/util/text.ts
export const LINE_SEPARATOR = "\n";

/**
 * 获取字符串的骨架。
 *
 * 通过去除字符串中的`{`, `}`, 数字（包括符号、小数）实现。
 */
export function getTextSkeleton(text: string): string {
    return text.replace(/[{}\d.+-]/gu, "");
}

File: go/translator/zh2en/util/text.go
package util

import (
	"regexp"
)

// LINE_SEPARATOR 行分隔符
const LINE_SEPARATOR = "\n"

// GetTextSkeleton 获取文本骨架
//
// 移除所有数字、空格、点、加号、减号
func GetTextSkeleton(text string) string {
	re := regexp.MustCompile(`[{}0-9\s.+-]`)
	return re.ReplaceAllString(text, "")
}
